src/common/components/hosts/utils.ts

```diff
@@ -71,6 +68,12 @@ export const canEditHost = (clusterStatus: Cluster['status'], status: Host['stat
     'pending-for-input',
   ].includes(status);
 
+export const canEditRole = (
+  cluster: Cluster,
```

Comment:
adding isSNO as a third optional parameter would be IMHO preferable. Avoiding passing whole cluster helps with reuse of this function on CIM side.

Reply:
ok! change it!